code-orange/VisitedLocationPanel.swift

```diff
@@ -0,0 +1,233 @@
+  private var currentLocationIndex = 0
+  private var locations: [RecordedLocation]
+
+  init?(locations: [RecordedLocation]) {
+    guard !locations.isEmpty else { return nil }
+    self.locations = locations
+    super.init(frame: .zero)
+    commonInit()
+  }
+  
+  required init?(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+  
+  private func commonInit() {
+    addSubview(mainStackView)
+    mainStackView.fillSuperview()
+    reloadData()
+  }
+  
+  private lazy var mainStackView: UIStackView = {
+    let stackView = UIStackView()
+    stackView.translatesAutoresizingMaskIntoConstraints = false
+    stackView.axis = .vertical
+    stackView.spacing = 16
+    stackView.addArrangedSubview(titleAndButtonsStackViewContainer)
+    stackView.addArrangedSubview(topLabelStack)
+    stackView.addArrangedSubview(bottomLabelStack)
+    stackView.addArrangedSubview(actionsStackView)
+    return stackView
+  }()
+  
+  private lazy var titleStackView: UIStackView = {
+    let stackView = UIStackView()
+    stackView.axis = .vertical
+    stackView.spacing = 2
+    stackView.alignment = .center
+    stackView.addArrangedSubview(pagesLabel)
+    stackView.addArrangedSubview(addressLabel)
+    stackView.addArrangedSubview(timeframeLabel)
+    return stackView
+  }()
+
+  private lazy var titleAndButtonsStackViewContainer: UIView = {
+    let stackView = UIStackView()
+    stackView.backgroundColor = .nxGrey10
+    stackView.axis = .horizontal
+    stackView.spacing = 2
+    stackView.alignment = .center
+    stackView.isLayoutMarginsRelativeArrangement = true
+    stackView.layoutMargins = UIEdgeInsets(top: 12, left: 16, bottom: 16, right: 16)
+    stackView.addArrangedSubview(leftPageButton)
+    stackView.addArrangedSubview(titleStackView)
+    stackView.addArrangedSubview(rightPageButton)
+    let view = UIView()
+    view.backgroundColor = .nxGrey10
+    view.addSubview(stackView)
+    stackView.translatesAutoresizingMaskIntoConstraints = false
+    stackView.fillSuperview()
+    return view
+  }()
+
+  private lazy var actionsStackView: UIStackView = {
+    let stackView = UIStackView()
+    stackView.axis = .horizontal
+    stackView.distribution = .fillEqually
+    stackView.spacing = 24
+    stackView.alignment = .center
+    stackView.isLayoutMarginsRelativeArrangement = true
+    stackView.layoutMargins = UIEdgeInsets(top: 12, left: 24, bottom: 24, right: 24)
+    stackView.addArrangedSubview(healthAdministrationWebsiteButton)
+    stackView.addArrangedSubview(callEmergencyServiceButton)
+    return stackView
+  }()
+
+  private lazy var leftPageButton: UIButton = {
+    let button = UIButton()
+    button.setImage(UIImage(named: "left"), for: .normal)
+    button.addTarget(self, action: #selector(previousLocationTapped), for: .touchUpInside)
+    return button
+  }()
+
+  private lazy var rightPageButton: UIButton = {
+    let button = UIButton()
+    button.setImage(UIImage(named: "right"), for: .normal)
```

Comment:
should we have these localised?

Reply:
Let's add it when we add localization 🙏

Reply:
we already added localization =)

Reply:
String localization I mean (i.e. default to English and fix LTR flows) 🌮